modules/color-picker/react/lib/ColorInput.tsx

```diff
@@ -44,23 +44,44 @@ const CustomHexInput = styled(TextInput)<Pick<ColorInputProps, 'disabled' | 'gro
     },
   ({disabled}) => ({
     backgroundColor: disabled ? colors.soap200 : '',
+  }),
+  ({theme}) => ({
+    paddingLeft:
+      theme.direction === ContentDirection.LTR ? '46px' : 'calc(100% - 86px) /* @noflip */',
+    // We're using @noflip because ColorInput should stay LTR, therefore, we need to adjust the padding using ContentDirection, not using rtl-css-js.
   })
 );
 
-const ColorInputContainer = styled('div')({
-  position: 'relative',
-});
+const ColorInputContainer = styled('div')<Pick<ColorInputProps, 'grow'>>(
+  {
+    position: 'relative',
+    width: colorInputWidth,
+  },
+  ({grow}) =>
```

Comment:
Nit: This isn't hurting anything, but why is this necessary? Block level elements are naturally 100% width. If it's not needed, you can remove it from the render function as well.

Reply:
It is necessary because the width is `colorInputWidth` otherwise (line 58)

Reply:
Ah - I missed that you added that as well, sorry. That way it behaves like the other input elements 👍